### ISSUE TITLE
Add asFlowWithContext to Java DSL

### ISSUE THREAD

**Issue by jmnght** [akka/akka#26884]
Java DSL is missing asFlowWithContext ops. Creating an issue as suggested by @raboof 
This would allow doing something like this without Graph:

```public Graph<FlowShape<ClassA, ClassD>, NotUsed> process() {

    return GraphDSL.create(b -> {

        UniformFanOutShape<ClassA, ClassA> A = b.add(Broadcast.create(2));
        FlowShape<ClassA, ClassB> B = b.add(Flow.of(ClassA.class).via(mapAtoB()));
        FlowShape<ClassB, ClassC> C = b.add(Flow.of(ClassB.class).via(mapBtoC)));
        FanInShape2<ClassA, ClassC, Pair<ClassA, ClassC>> D = b.add(Zip.create());

        Flow<Pair<ClassA, ClassC>, Pair<ClassA, ClassC>, NotUsed> AC = Flow.create();
        FlowShape<Pair<ClassA, ClassC>, ClassD> F = b.add(AC.via(mapACtoD()));

        b.from(A).toInlet(D.in0());
        b.from(A).via(B).via(C).toInlet(D.in1());
        b.from(D.out()).via(F);

        return new FlowShape<ClassA, ClassD>(B.in(), F.out());
        });
    }
}
```

**Comment by johanandren** [comment]
Thanks for reporting, would you be up for submitting a PR, should be fairly easy to do, delegate to the Scala one and use Java lambda types like other JavaDSL factories does, the JavaDSL `FlowWithContext` is already there, so I think it is just the factory missing.

**Comment by fagossa** [comment]
@johanandren @jmnght can I take this? I have some spare time this weekend

**Comment by jmnght** [comment]
@fagossa go ahead. I don’t think I can get to it next week

**Comment by johanandren** [comment]
Just FYI there were some other issue around with-context that was discovered last week: #26896 May or may not be relevant to this issue.

**Comment by contrun** [comment]
This issue seems to be resolved. https://github.com/akka/akka/blob/0e4d41ad33dbeb00b598cb75b4d29899371bdc8c/akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala#L3817-L3820

**Comment by ennru** [comment]
Indeed, fixed in Akka 2.5.22:

https://github.com/akka/akka/commit/4ee1ae09a56a5accf8b539fc6e198ca00a4e239a
